src/probe/scene_detect.rs

```diff
@@ -90,38 +94,60 @@ pub fn detect_scene<S: ::std::hash::BuildHasher>(
               error!("Error : unexpected detection on stream ${index}");
               break;
             }
-            let detected_scene = streams[(index) as usize].detected_scene.as_mut().unwrap();
-            let detected_false_scene = streams[(index) as usize]
-              .detected_false_scene
-              .as_mut()
-              .unwrap();
-            let mut scene = SceneResult {
-              frame_index: 0,
-              score: 0,
-              scene_number: 0,
-            };
-            let mut false_scene = FalseSceneResult { frame: 0 };
 
             if let Some(value) = entry_map.get("lavfi.scd.time") {
-              scene.frame_index = (value.parse::<f32>().unwrap() * frame_rate) as i64;
+              let detected_scene = streams[(index) as usize].detected_scene.as_mut().unwrap();
+              // let detected_false_scene = streams[(index) as usize]
+              //   .detected_false_scene
+              //   .as_mut()
+              //   .unwrap();
+              let mut scene = SceneResult {
+                frame_start: 0,
+                frame_end: stream_frames,
+                frames_length: 0,
+                score: 0,
+                index: 0,
+              };
+              // let mut false_scene = FalseSceneResult { index: 0 };
+
+              scene.frame_start = (value.parse::<f32>().unwrap() * video_details.frame_rate) as i64;
               if let Some(value) = entry_map.get("lavfi.scd.score") {
                 scene.score = (value.parse::<f32>().unwrap()) as i32;
               }
-
-              if let Some(last_detect) = detected_scene.last() {
-                if scene.frame_index - last_detect.frame_index <= 1 {
-                  false_scene.frame = scene.frame_index;
-                  detected_false_scene.push(false_scene);
-                }
+              if let Some(last_detect) = detected_scene.last_mut() {
+                last_detect.frame_end = scene.frame_start - 1;
+                last_detect.frames_length = last_detect.frame_end - last_detect.frame_start + 1;
+                // if scene.frame_start - last_detect.frame_start <= 1 {
+                //   false_scene.index = scene.frame_start;
+                //   detected_false_scene.push(false_scene);
+                // }
               }
 
               scene_count += 1;
-              scene.scene_number = scene_count;
+              scene.index = scene_count;
               detected_scene.push(scene);
             }
           }
         }
       }
+      for index in video_indexes.clone() {
+        let detected_scene = streams[(index) as usize].detected_scene.as_mut().unwrap();
+        if !detected_scene.is_empty() {
+          detected_scene.insert(
```

Comment:
je mettrais plutôt un init au début (dans le premier `for index in video_indexes`)